meerk40t/gui/simulation.py

```diff
@@ -461,63 +589,301 @@ def on_mouse_right_down(self, event=None):
+                #         setvalue = 0
+                #         if len(params) > 0:
+                #             try:
+                #                 mask = int(params[0])
+                #             except ValueError:
+                #                 mask = 0
+                #         if len(params) > 1:
+                #             try:
+                #                 setvalue = int(params[1])
+                #             except ValueError:
+                #                 setvalue = 0
+                #         if mask != 0 or setvalue != 0:
+                #             addop = OutputOperation(mask, setvalue)
+                elif optype == "util goto":
+                    if opparam is not None:
+                        params = opparam.split(",")
+                        x = 0
+                        y = 0
+                        if len(params) > 0:
+                            try:
+                                x = float(Length(params[0]))
+                            except ValueError:
+                                x = 0
+                        if len(params) > 1:
+                            try:
+                                y = float(Length(params[1]))
+                            except ValueError:
+                                y = 0
+                        addop = GotoOperation(x=x, y=y)
+                elif optype == "util origin":
+                    if opparam is not None and opparam != "":
+                        params = opparam.split(",")
+                        x = 0
+                        y = 0
+                        if len(params) > 0:
+                            try:
+                                x = float(Length(params[0]))
+                            except ValueError:
+                                x = 0
+                        if len(params) > 1:
+                            try:
+                                y = float(Length(params[1]))
+                            except ValueError:
+                                y = 0
+                        addop = SetOriginOperation(x=x, y=y)
+                    else:
+                        addop = SetOriginOperation(x=None, y=None)
+                elif optype == "util wait":
+                    if opparam is not None:
+                        try:
+                            opparam = float(opparam)
+                        except ValueError:
+                            opparam = None
+                    if opparam is not None:
+                        addop = WaitOperation(wait=opparam)
+                if addop is not None:
+                    pre_items.append([desc, addop])
+
+        menu.AppendSeparator()
+        for entry in pre_items:
+            self.Bind(
+                wx.EVT_MENU,
+                insert_operation(entry[1]),
+                menu.Append(
+                    wx.ID_ANY,
+                    _("Insert '{operation}' to cutplan").format(operation=entry[0]),
```

Comment:
Maybe insert before or something. It seems like you have insert at the end and insert before this selected operation. It's probably fine.